test/test.py

```diff
@@ -78,19 +71,17 @@ def tearDown(self):
         os.environ = self._os_environ
 
     def run_completer(self, parser, command, point=None, completer=autocomplete, **kwargs):
-        command = ensure_str(command)
-
         if point is None:
             point = str(len(command))
         with TemporaryFile() as t:
-            os.environ["COMP_LINE"] = ensure_bytes(command) if USING_PYTHON2 else command
+            os.environ["COMP_LINE"] = command
             os.environ["COMP_POINT"] = point
             with self.assertRaises(SystemExit) as cm:
                 completer(parser, output_stream=t, exit_method=sys.exit, **kwargs)
             if cm.exception.code != 0:
                 raise Exception("Unexpected exit code %d" % cm.exception.code)
             t.seek(0)
-            return t.read().decode(sys_encoding).split(IFS)
+            return t.read().decode().split(IFS)
```

Comment:
Since this is in the test suite, I'm not going to do anything about it (I'm assuming the test suite will always run under the standard encoding, utf-8).

Reply:
Never mind, since this hooks up to the output stream of the completer, it also needs to be changed to use text mode.